app/controllers/volunteers_controller.rb

```diff
@@ -25,20 +33,34 @@ def create
   end
 
   def update
+    authorize @volunteer
     if @volunteer.update(volunteer_params)
-      redirect_to @volunteer, notice: t('volunteer_updated')
+      if @state == Volunteer::INTERESTED && @volunteer.state == Volunteer::ACCEPTED
```

Comment:
`@volunteer.state` contains the state after the update. `@state` contains the state before the update. We need to check, if the state has changed from interested to accepted. How can I check that by only using `@volunteer.state'?